test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+  distinct iterations of a single data-parallel loop would violate the
+  order-independent property and not be a legal use of the loop form
+  (using a task-parallel loop in such cases would be more appropriate,
+  as we will see).
+
+  foreach-loops
+  -------------
+
+  The first, and simplest, data-parallel loop is the ``foreach`` loop.
+  This loop form asserts that the loop meets the order-independent and
+  unsynchronized properties above, and specifies that its iterations
+  should be implemented using hardware parallelism if possible.  When
+  executing a foreach-loop on a conventional processor or GPU, the
+  compiler will attempt to implement its iterations using any
+  SIMD/SIMT parallelism that's available.  For example, if executing
+  the loop on a processor with vector instructions, it will attempt to
+  implement the loop using those instructions if possible.  Notably, a
+  foreach-loop will not implement its iterations using multiple Chapel
+  tasks or software threads (the forall-loop, below, does this).
+
+  Syntactically, foreach-loops are identical to for-loops, simply
+  using the ``foreach`` keyword.  For example, the following
+  ``foreach`` loop will double the values of the array ``A`` declared
+  above:
+
+*/
+
+  foreach a in A do
+    a *= 2;
+
+  writeln("After our first foreach-loop, A is: ", A);
+
+/*
+
+  Because each iteration is performing its own operations on its own
+  elements of A, this loop is trivially parallel-safe, and completely
+  reasonable to write using ``foreach``.  When running the computation
+  on a processor that supports vector instructions for performing
+  floating point multiplications, the ``foreach`` keyword's assertion
+  that the loop is legal to parallelize improves the compiler's
+  ability to implement the loop using those instructions.
+
+  Like the for-loops above, Chapel's foreach-loops support zippered
+  iteration.  For example, the following loop performs a zippered
+  iteration over the array `A` and an unbounded range `1..`.
+
+*/
+
+  foreach (a, i) in zip(A, 1..) do
+    a += (i / 100.0);
+
+   writeln("After our first zippered foreach-loop, A is: ", A);
+
+/*
+
+  Note that if a Chapel program only used foreach-loops to express its
+  parallelism, it would never make use of the multiple processor cores
+  of a modern processor nor the distinct compute nodes of a cluster or
+  HPC system.  This is because foreach-loops don't ever introduce new
+  Chapel tasks, and tasks are the only way to run in parallel at
+  system scales beyond a single processor.  As a result, to leverage
+  the full power of most parallel platforms, we need to look to
+  Chapel's other parallel loop forms.
+
+
+  forall-loops
+  ------------
+
+  Forall-loops are similar to foreach-loops, except that they have the
+  potential to be implemented using multiple Chapel tasks.  This
+  permits them to use multiple cores and/or compute nodes to execute
+  the loop's iterations.
+
```

Comment:
Here is where I suggest you show an example forall loop.  Give example code early and often.